### PR TITLE
Move PATH editing logic to JS

### LINKED ISSUE #112: yvm use and yvm exec - behaviour for getting config differ
**Describe the bug**
yvm use, looks in the current directory for a `.yvmrc` file. yvm exec does a more indepth tree and file search

**Expected behavior**
They should behave the same

### PR DESCRIPTION
**Pull request by francoiscampbell**
## Description

- Changes `yvm use` to use `yvm get-new-path` under the hood, moving most of the logic into JS for consistency
- Closes #112

### DevQA Steps
- `make install-watch`
- Open a new shell
- `yvm use` should still work
- `yvm get-new-path 1.7.0` should return a PATH string for yarn 1.7.0
- `yvm get-new-path` should use the version in .yvmrc
- `yvm use 1.6.0` then `yvm get-new-path 1.7.0` should edit the existing PATH string, not prepend a new segment to it

### CODE REVIEW

File: src/yvm.js [hunk 1]
```diff
@@ -98,6 +98,18 @@ argParser
         listVersions()
     })
 
+argParser
+    .command('get-path [version]')
+    .description('Internal command: Gets a new PATH string for "yvm use"')
+    .action(version => {
+        if (!version) {
```
Comment (jakebolam) [inline src/yvm.js]: Can we look in the configPath for the version too (i.e. version is optional)
```
        // TODO: says whether installed or not
        if (version) {
            const path = getExtractionPath(version)
            log.capturable(path)
            return
        }

        const configVersion = getRcFileVersion()
        if (configVersion) {
            const path = getExtractionPath(configVersion)
            log.capturable(path)
            return
        }
```

File: src/yvm.js [hunk 2]
```diff
@@ -98,6 +98,18 @@ argParser
...
+            log('Version missing')
```
Comment (jakebolam) [inline src/yvm.js]: Can we exit non-zero?
```
        log('No yarn version found or supplied')
        process.exit(1)
?
```

File: src/yvm.sh [hunk 3]
```diff
@@ -4,34 +4,13 @@ command=$1
...
-        yvm_err 'version is required'
-        return 3
-    else yvm_echo "$(yvm_version_dir)/v${VERSION}"
+    local PROVIDED_VERSION=${1}
+    NEW_PATH=$(yvm get-path ${PROVIDED_VERSION})
+    if [ -z "${NEW_PATH}" ]; then
+        yvm_err "Could not get new path from yvm"
+    else
+        PATH=${NEW_PATH}
+        yvm_echo "Set yarn version to ${PROVIDED_VERSION}"
```
Comment (jakebolam) [inline src/yvm.sh]: [nb] Drop the provided version (as it's hard to get from when yvm finds it from the rc file), `Set yarn to path blah`

File: src/commands/getPath.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const path = require('path')
+
+const { versionRootPath, yvmPath } = require('../common/utils')
+
+const getPath = (version, rootPath = yvmPath) => {
```
Comment (jakebolam) [inline src/commands/getPath.js]: Can we get a unit tests. This happen to me earlier, would be good if we could extend this to somehow now about global yarn, and remove it too:
```
➜  fe-libraries git:(feature/add-jwt) ✗ yvm use
Found config /code/fe-libraries/.yvmrc
Set yarn to /Users/jakebolam/.yvm/versions/v1.7.0
➜  fe-libraries git:(feature/add-jwt) ✗ $PATH
zsh: not a directory: /Users/jakebolam/.yvm/versions/v1.7.0/bin/yarn/bin:/Users/jakebolam/.nvm/versions/node/v8.11.3/bin:/Users/jakebolam/.yarn/bin:/Users/jakebolam/.config/yarn/global/node_modules/.bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
➜  fe-libraries git:(feature/add-jwt) ✗ yarn --version
1.0.2
```

File: src/yvm.js [hunk 5]
```diff
@@ -98,6 +98,20 @@ argParser
         listVersions()
     })
 
+argParser
+    .command('get-path [version]')
+    .description('Internal command: Gets a new PATH string for "yvm use"')
+    .action(withRcFileVersion(version => {
```
Comment (jakebolam) [inline src/yvm.js]: Does this work with a provided version from action?

Comment (jakebolam) [reply src/yvm.js]: Little confused about this, would it be better to make the code more explicit? Instead of hiding nested functions

Comment (francoiscampbell) [reply src/yvm.js]: yup, the same is used in `yvm install`

File: src/commands/getPath.js (hunk 4 above)
Comment (francoiscampbell) [reply src/commands/getPath.js]: yeah, i'm thinking a shell test or two in addition to unit tests for each command

File: src/yvm.sh [hunk 6]
```diff
@@ -4,34 +4,12 @@ command=$1
...
-
-yvm_version_path() {
-    local VERSION
-    VERSION="${1-}"
-    if [ -z "${VERSION}" ]; then
-        yvm_err 'version is required'
-        return 3
-    else yvm_echo "$(yvm_version_dir)/v${VERSION}"
+    local PROVIDED_VERSION=${1}
+    NEW_PATH=$(yvm get-path ${PROVIDED_VERSION})
```
Comment (jakebolam) [inline src/yvm.sh]: `yvm` ? does this work?

Comment (francoiscampbell) [reply src/yvm.sh]: it does, but i can change it to use node directly

File: src/yvm.js (hunk 5 above)
Comment (francoiscampbell) [reply src/yvm.js]: `withRcFileVersion` is actually kinda heavy:

```
const withRcFileVersion = action => (maybeVersionArg, ...rest) => {
    if (maybeVersionArg) {
        if (isValidVersionString(maybeVersionArg)) {
            log(`Using provided version: ${maybeVersionArg}`)
            action(maybeVersionArg, ...rest)
            return
        }
    }

    rest.unshift(maybeVersionArg)

    try {
        const version = getRcFileVersion()
        validateVersionString(version)
        log(`Using yarn version: ${version}`)
        action(version, ...rest)
    } catch (e) {
        log(e.message)
        process.exit(1)
    }
}
```

Comment (francoiscampbell) [reply src/yvm.js]: it has to support extra args, etc

File: src/yvm.sh (hunk 6 above)
Comment (dat2) [reply src/yvm.sh]: 🐔 and 🥚

**Pull request merged**